Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PaalmanPingsMonteCarloAbsorption.py

```diff
@@ -0,0 +1,906 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import (DataProcessorAlgorithm, AlgorithmFactory, PropertyMode, WorkspaceGroupProperty, SpectraAxis,
+                        WorkspaceGroup, WorkspaceProperty, Sample)
+from mantid.kernel import (VisibleWhenProperty, PropertyCriterion, StringListValidator, IntBoundedValidator,
+                           FloatBoundedValidator, Direction, LogicOperator, EnabledWhenProperty)
+from mantid.simpleapi import (config, logger, mtd)
+import math
+import numpy as np
+import os.path
+
+
+class PaalmanPingsMonteCarloAbsorption(DataProcessorAlgorithm):
+    # General variables
+    _emode = None
+    _efixed = None
+    _general_kwargs = None
+    _shape = None
+    _height = None
+    _isis_instrument = None
+
+    # Sample variables
+    _sample_angle = None
+    _sample_center = None
+    _sample_chemical_formula = None
+    _sample_density = None
+    _sample_density_type = None
+    _sample_inner_radius = None
+    _sample_outer_radius = None
+    _sample_radius = None
+    _sample_thickness = None
+    _sample_unit = None
+    _sample_width = None
+    _sample_ws = None
+
+    # Container variables
+    _container_angle = None
+    _container_center = None
+    _container_chemical_formula = None
+    _container_density = None
+    _container_density_type = None
+    _container_radius = None
+    _container_inner_radius = None
+    _container_outer_radius = None
+    _container_thickness = None
+    _container_width = None
+
+    # Output workspaces
+    _ass_ws = None
+    _acc_ws = None
+    _assc_ws = None
+    _acsc_ws = None
+    _output_ws = None
+
+    def category(self):
+        return "Workflow\\Inelastic;CorrectionFunctions\\AbsorptionCorrections;Workflow\\MIDAS"
+
+    def seeAlso(self):
+        return ["MonteCarloAbsorption", "ApplyPaalmanPingsCorrection"]
+
+    def summary(self):
+        return "Calculates absorption corrections in Paalman & Pings formalism for a given sample " \
+               "and optionally its environment using a Monte Carlo simulation."
+
+    def checkGroups(self):
+        return False
+
+    def PyInit(self):
+        # Sample Input
+        self.declareProperty(WorkspaceProperty('SampleWorkspace', '', direction=Direction.Input),
+                             doc='Sample Workspace')
```

Comment:
Perhaps would be helpful to expand the doc string to explain this workspace contains measurement with sample and container present.